marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadFunctionMetaDataCommand.java

```diff
@@ -0,0 +1,76 @@
+package com.marklogic.hub.deploy.commands;
+
+import com.marklogic.appdeployer.command.AbstractCommand;
+import com.marklogic.appdeployer.command.CommandContext;
+import com.marklogic.appdeployer.command.SortOrderConstants;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.datamovement.ApplyTransformListener;
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.QueryBatcher;
+import com.marklogic.client.document.ServerTransform;
+import com.marklogic.client.query.StructuredQueryBuilder;
+import com.marklogic.hub.DatabaseKind;
+import com.marklogic.hub.HubConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class LoadFunctionMetaDataCommand extends AbstractCommand {
+    @Autowired
+    private HubConfig hubConfig;
+    private boolean forceLoad = true;
+
+    public LoadFunctionMetaDataCommand() {
+        super();
+        // This command has to be executed after modules are loaded
+        setExecuteSortOrder(SortOrderConstants.LOAD_MODULES + 1);
+    }
+    public void setForceLoad(boolean forceLoad) {
+        this.forceLoad = forceLoad;
+    }
+
+
+    @Override
+    public void execute(CommandContext context) {
+        DatabaseClient databaseClient = hubConfig.newStagingClient(hubConfig.getDbName(DatabaseKind.MODULES));
+        DataMovementManager dataMovementManager = databaseClient.newDataMovementManager();
+
+        StructuredQueryBuilder sb = databaseClient.newQueryManager().newStructuredQueryBuilder();
+
+
+        ServerTransform serverTransform = new ServerTransform("ml:loadFunctionMetaData");
+        serverTransform.addParameter("forceLoad", String.valueOf(forceLoad));
+
+        ApplyTransformListener transformListener = new ApplyTransformListener()
+            .withTransform(serverTransform)
+            .withApplyResult(ApplyTransformListener.ApplyResult.IGNORE)
+            .onBatchFailure((batch, throwable) -> {
+                for(String item: batch.getItems()) {
+                    logger.error("Unable to load function metadata for: " + item);
+                }
+                logger.error(throwable.getMessage());
```

Comment:
@ryanjdew @aebadirad  Thoughts on whether a failure to generate function metadata should result in the deployment failing? I lean towards making it fail. ml-gradle has a property to tell it to catch exceptions and keep going in case that's a problem. 

@srinathgit If we do agree on this throwing an error, then I think you'll need to declare a Throwable variable before you construct this listener. onBatchFailure can then assign the throwable that it catches to this declared variable. After DMSDK finishes, if the throwable is null, then throw it to cause a hard failure.

Reply:
I agree with making it fail. My experience is people tend to ignore logs and march on, only to have more issues down the road.